dotCMS/src/main/java/com/dotcms/security/apps/AppsAPIImpl.java

```diff
@@ -681,28 +688,60 @@ private void invalidateCache() {
         }
     }
 
-    private static String getServiceDescriptorDirectory() {
+    /**
+     * This is the directory intended for customers use
+     * @return
+     */
+    private static Path getUserAppsDescriptorDirectory() {
         final Supplier<String> supplier = () -> APILocator.getFileAssetAPI().getRealAssetsRootPath()
         + File.separator + SERVER_DIR_NAME + File.separator + APPS_DIR_NAME + File.separator;
         final String dirPath = Config
                 .getStringProperty(APPS_DIR_PATH_KEY, supplier.get());
-        return Paths.get(dirPath).normalize().toString();
+        return Paths.get(dirPath).normalize();
     }
 
-    private Set<String> listAvailableYamlFiles() throws IOException {
-        final String ymlFilesPath = getServiceDescriptorDirectory();
-        final File basePath = new File(ymlFilesPath);
+    /**
+     * This is the Apps-System-Folder which is meant to hold system apps.
+     * Those that can not be override and are always available.
+     * @return
+     */
+    static Path getSystemAppsDescriptorDirectory() throws URISyntaxException, IOException {
+        final URL res = AppsAPIImpl.class.getClassLoader().getResource("apps");
```

Comment:
Codacy found an issue: [In J2EE, getClassLoader() might not work as expected.  Use Thread.currentThread().getContextClassLoader() instead.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5903396)